db/dcrpg/internal/vinoutstmts.go

```diff
@@ -58,7 +61,16 @@ const (
 	SelectFundingOutpointByVinID = `SELECT prev_tx_hash, prev_tx_index, prev_tx_tree FROM vins WHERE id=$1;`
 	SelectFundingTxByVinID       = `SELECT prev_tx_hash FROM vins WHERE id=$1;`
 	SelectSpendingTxByVinID      = `SELECT tx_hash, tx_index, tx_tree FROM vins WHERE id=$1;`
-	SelectAllVinInfoByID         = `SELECT * FROM vins WHERE id=$1;`
+	SelectAllVinInfoByID         = `SELECT id, tx_hash, tx_index, tx_tree, is_valid, block_time,
+		prev_tx_hash, prev_tx_index, prev_tx_tree, value_in FROM vins WHERE id = $1;`
+	SetIsValidByTxHash = `UPDATE vins SET is_valid = $1 WHERE tx_hash = $2;`
+
+	SetVinsTableCoinSupplyUpgrade = `UPDATE vins SET is_valid = $1, block_time = $2, value_in = $3
+		WHERE tx_hash = $4 and tx_index = $5 and tx_tree = $6;`
+
+	SelectCoinSupply = `SELECT block_time, sum(value_in) FROM vins WHERE
```

Comment:
A comment explaining that this sums the generated coins for all stakebase and only not-invalidated coinbase transactions.

Reply:
Hmm, which reminds me, it may be complicated if we want to hang on to orphaned blocks in side chains.  For that we'll need a new column called `main_chain` (or similar) and this query will need to include a condition that `mainchain = TRUE`.  This is a note to myself.

Reply:
Where is the `main_chain` be set from? What is its relationship with orphaned blocks?

Reply:
This would be part of reorg handling.  orphaned blocks would have main_chain=false